pkg/skaffold/gcbreposv2/repo_resolver_test.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2024 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcbreposv2
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"cloud.google.com/go/cloudbuild/apiv2/cloudbuildpb"
+	"github.com/googleapis/gax-go/v2"
+
+	"github.com/GoogleContainerTools/skaffold/v2/testutil"
+)
+
+type GCBReposClientMock struct {
+	Region              string
+	Project             string
+	Connection          string
+	Repo                string
+	GitRepo             string
+	ReadToken           string
+	ShouldErrorGetRepo  bool
+	ShouldErrorGetToken bool
+}
+
+func (c GCBReposClientMock) GetRepository(ctx context.Context, req *cloudbuildpb.GetRepositoryRequest, opts ...gax.CallOption) (*cloudbuildpb.Repository, error) {
+	if c.ShouldErrorGetRepo {
+		return nil, fmt.Errorf("failed to get repo")
+	}
+
+	if req.Name == fmt.Sprintf("projects/%v/locations/%v/connections/%v/repositories/%v", c.Project, c.Region, c.Connection, c.Repo) {
+		return &cloudbuildpb.Repository{
+			Name:      c.Repo,
+			RemoteUri: c.GitRepo,
+		}, nil
+	}
+
+	return nil, fmt.Errorf("invalid request")
+}
+
+func (c GCBReposClientMock) FetchReadToken(ctx context.Context, req *cloudbuildpb.FetchReadTokenRequest, opts ...gax.CallOption) (*cloudbuildpb.FetchReadTokenResponse, error) {
+	if c.ShouldErrorGetToken {
+		return nil, fmt.Errorf("failed to get token")
+	}
+
+	if req.Repository == fmt.Sprintf("projects/%v/locations/%v/connections/%v/repositories/%v", c.Project, c.Region, c.Connection, c.Repo) {
+		return &cloudbuildpb.FetchReadTokenResponse{
+			Token: c.ReadToken,
+		}, nil
+	}
+	return nil, fmt.Errorf("invalid request")
+}
+
+func (c GCBReposClientMock) Close() error { return nil }
+
+func TestGetRepoURI(t *testing.T) {
+	tests := []struct {
+		description         string
+		gcpRegion           string
+		gcpProject          string
+		gcpConnection       string
+		gcpRepo             string
+		expectedGitRepo     string
+		expectedToken       string
```

Comment:
no non-zero values for test

Reply:
I changed the structure of the unit tests, also taking into account that now this package has the logic to build the oauth2 format. Let me know how it looks. Thanks!